src/modules/cpu/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include <omp.h>
+
+Rpp32f getSquare(Rpp32f &value)
+{
+    return (value * value);
+}
+
+Rpp32f getMax(Rpp32f *values, Rpp32s srcLength)
+{
+    Rpp32f max = values[0];
+    for(int i = 1; i < srcLength; i++)
+        max = std::max(max, values[i]);
+    return max;
+}
+
+RppStatus non_silent_region_detection_host_tensor(Rpp32f *srcPtr,
+                                                  RpptDescPtr srcDescPtr,
+                                                  Rpp32s *srcLengthTensor,
+                                                  Rpp32f *detectedIndexTensor,
+                                                  Rpp32f *detectionLengthTensor,
+                                                  Rpp32f cutOffDB,
+                                                  Rpp32s windowLength,
+                                                  Rpp32f referencePower,
+                                                  Rpp32s resetInterval,
+                                                  rpp::Handle& handle)
+{
+    Rpp32u numThreads = handle.GetNumThreads();
+    const Rpp32f cutOff = std::pow(10.0f, cutOffDB * 0.1f);
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        Rpp32s srcLength = srcLengthTensor[batchCount];
+        Rpp32f *mmsBuffer = static_cast<Rpp32f *>(calloc(srcLength, sizeof(Rpp32f)));
```

Comment:
Please add a comment here stating why specifically srcLength could be a wide range, and its hard to statically initialize or pre-initialize during handle creation. You can even give a range

Reply:
Done
Again somehow this change is not being shown in github ui